modules/aruco/samples/tutorial_charuco_create_detect.cpp

```diff
@@ -0,0 +1,165 @@
+//! [charucohdr]
+#include <opencv2/aruco/charuco.hpp>
+//! [charucohdr]
+#include <opencv2/highgui.hpp>
+#include <iostream>
+#include <string>
+
+namespace {
+const char* about = "A tutorial code on charuco board creation and detection of charuco board with and without camera caliberation";
+const char* keys  =
+        "{c        |       | Put value of c=1 to create charuco board;\nc=2 to detect charuco board without camera calibration;\nc=3 to detect charuco board with camera calibration and Pose Estimation}";
+}
+
+void createBoard();
+void detectCharucoBoardWithCalibrationPose();
+void detectCharucoBoardWithoutCalibration();
+
+
+
+
+static bool readCameraParameters(std::string filename, cv::Mat &camMatrix, cv::Mat &distCoeffs) {
+    cv::FileStorage fs(filename, cv::FileStorage::READ);
+    if(!fs.isOpened())
+        return false;
+    fs["camera_matrix"] >> camMatrix;
+    fs["distortion_coefficients"] >> distCoeffs;
+    return true;
+}
+
+
+void createBoard(){
+    cv::Ptr<cv::aruco::Dictionary> dictionary = cv::aruco::getPredefinedDictionary(cv::aruco::DICT_6X6_250);
+    //! [createBoard]
+    cv::Ptr<cv::aruco::CharucoBoard> board = cv::aruco::CharucoBoard::create(5, 7, 0.04f, 0.02f, dictionary);
+    cv::Mat boardImage;
+    board->draw( cv::Size(600, 500), boardImage, 10, 1 );
+    //! [createBoard]
+    cv::imwrite("BoardImage.jpg",boardImage);
+}
+
+//! [detwcp]
+void detectCharucoBoardWithCalibrationPose(){
+    cv::VideoCapture inputVideo;
+    inputVideo.open(0);
+    //! [matdiscoff]
+    cv::Mat cameraMatrix, distCoeffs;
+    std::string filename="calib.txt";
+    bool readOk=readCameraParameters(filename,cameraMatrix, distCoeffs);
+    //! [matdiscoff]
+    if(!readOk) {
+            std::cerr << "Invalid camera file" << std::endl;
+    }else{
+    //! [dictboard]
+    cv::Ptr<cv::aruco::Dictionary> dictionary = cv::aruco::getPredefinedDictionary(cv::aruco::DICT_6X6_250);
```

Comment:
Please fix indentation (4 spaces).
You may use `clang-format` for that (for this file only):
```
clang-format --style=WebKit --sort-includes=0 -i modules/aruco/samples/tutorial_charuco_create_detect.cpp
```

Reply:
Done